plutus-core/plutus-core/src/PlutusCore/Parser/Builtin.hs

```diff
@@ -7,32 +7,34 @@ module PlutusCore.Parser.Builtin where
 import Data.Text qualified as T
 import Data.Text.Internal.Read (hexDigitToInt)
 import PlutusPrelude (Word8)
-import Text.Megaparsec (MonadParsec (takeWhileP), choice, many, manyTill)
+import Text.Megaparsec (MonadParsec (takeWhileP), choice, customFailure, getSourcePos, many, manyTill)
 import Text.Megaparsec.Char (char, hexDigitChar)
 import Text.Megaparsec.Char.Lexer qualified as Lex hiding (hexadecimal)
 
 import Data.ByteString (pack)
 import Data.Map (Map, empty, insert, lookup)
 import PlutusCore.Default
+import PlutusCore.Error
 import PlutusCore.Parser.ParserCommon (Parser, isIdentifierChar, lexeme, symbol, whitespace)
 import PlutusCore.Parser.Type (defaultUniType)
 import PlutusCore.Pretty (display)
 import Prelude hiding (lookup)
 
-mkCachedBuiltin :: [DefaultFun] -> Map T.Text DefaultFun -> Map T.Text DefaultFun
-mkCachedBuiltin (hdFns:tlFns) builtinMap =
-    mkCachedBuiltin tlFns (insert (display hdFns) hdFns builtinMap)
-mkCachedBuiltin [] builtinMap = builtinMap
-
 cachedBuiltin :: Map T.Text DefaultFun
-cachedBuiltin = mkCachedBuiltin [minBound .. maxBound] empty
+cachedBuiltin
+  = foldl
+      (\ currMap fn -> insert (display fn) fn currMap)
+      empty
+      [minBound .. maxBound]
```

Comment:
Whenever I see `foldl`, I ask myself "why is it not `foldl'`?" and in this case there's no good reason for that. `foldl'` is the only [sensible](https://well-typed.com/blog/2014/04/fixing-foldl) default.

And if you make it `foldl'`, you're practically inlining `Map.fromList` here:

```haskell
fromList :: Ord k => [(k,a)] -> Map k a 
fromList xs       
  = foldlStrict ins empty xs
  where
    ins t (k,x) = insert k x t
```

at which point you could just use `Map.fromList`.

But this all is nitpicking and I think the current definition is fine.

Reply:
In fact, I think this is even a good list-comprehension moment!
```
Map.fromList [ (display fn, fn) | fn <- [minBound .. maxBound] ]
```